src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -401,22 +406,34 @@ public with sharing class BDI_MigrationMappingUtility {
             }
         }
 
+        // create and enqueue records to be created
         if (referenceFieldMappingsByStrings.size()>0){
             for (Data_Import_Field_Mapping__mdt referenceFieldMapping : referenceFieldMappingsByStrings.values()) {
                 Data_Import_Field_Mapping__mdt clone = referenceFieldMapping.clone();
 
                 DataImportFieldMapping fieldMapping = new DataImportFieldMapping(
                         clone, MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
-                fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
-                fieldMapping.setObjectMapping(
-                        referenceFieldMapping.DeveloperName,
-                        fieldMapping.dataImportObjectName);
 
-                fieldMapping.setRecordName(null, false);
-                queuedMetadataTypesForDeploy.add(fieldMapping.getMetadataRecord());
+                if (dataImportObjectDeveloperNameByLabel.size()>0){
+                    String existingObjectMappingDeveloperName =
+                            dataImportObjectDeveloperNameByLabel.get(fieldMapping.targetObjectMappingLabel);
+
+                    fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                    fieldMapping.setObjectMapping(
+                            existingObjectMappingDeveloperName,
```

Comment:
@callen-sfdo This line corrects the issue and sets the object name to the actual Migrated Custom Object Mapping record name (ex: "Payment_9sudflks9" instead of "Payment_Origin_Name"). The supporting helper class helps to obtain this value.